webportal/server/server.js

```diff
@@ -17,13 +17,28 @@
 
 
 // module dependencies
+const request = require('request');
 const config = require('./config/index');
 const logger = require('./config/logger');
 const app = require('./config/express');
 
 logger.info('config: %j', config);
 
-// start the server
-app.listen(config.serverPort, () => {
-  logger.info('Webportal server starts on port %d', config.serverPort);
-});
\ No newline at end of file
+// check the connection to rest server
+const checkRestServer = (callback) => {
+  request(config.restServer, (err) => {
+    if (err) {
+      logger.warn('Cannot connect to REST Server, retry after 5 seconds.')
+      setTimeout(checkRestServer, 5000, callback)
+    } else {
+      callback()
+    }
+  })
+};
+
+checkRestServer(() => {
+  // start the server
+  app.listen(config.serverPort, () => {
+    logger.info('Webportal server starts on port %d', config.serverPort);
+  });
+})
```

Comment:
Synced with Qixiang offline, we can add a "ping restserver" to workaround the problem on CI/CD bed, but we should still not introduce dependency on RestAPI here. Webportal has his own error handling and can give user correct response when RestServer is not avaliabile.